presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -411,10 +417,20 @@ void PrestoServer::run() {
   addAdditionalPeriodicTasks();
   periodicTaskManager_->start();
 
+  bool timesliceRunning = false;
+  std::thread timesliceThread;
+  if (FLAGS_task_timeslice_micros) {
+    timesliceThread = std::thread([&]() { timesliceLoop(timesliceRunning); });
+    timesliceRunning = true;
+  }
+
   // Start everything. After the return from the following call we are shutting
   // down.
   httpServer_->start(getHttpServerFilters());
-
+  if (timesliceRunning) {
+    timesliceRunning = false;
+    timesliceThread.join();
+  }
```

Comment:
@oerling From line 396 to line 430, it looks we're trying to make it more clean and informative with the server shutdown steps for debuggability, ex. by logging better with PRESTO_SHUTDOWN_LOG(). Would you like to put the stopping of the timesliceLoop thread into the same format?